### DISCUSSION TITLE
Filtering issues, Syntax?

### DISCUSSION THREAD

**Discussion by realJogicodes** [pocketbase/pocketbase#4882] [Q&A]
Hi, 

I am struggling implementing filters. I  think it's syntax related but cannot figure it out because I also think I am doing it right but I am not. Bad combination.

Maybe your eyes can see it?

Anyway, I have:

`pb.collection('entries').getFullList({expand: 'features'});`

My app holds `entries` and has a relation to `features` because the entries have features. It is a to-many relation. 
What I want to do, is pass in the `features` id string into the `entries` query and filter by it. Any syntax I tried gave me either and empty list or a 400 return.

```
console.log(filters); 
// logs this example for a filter string: 1pnq0aoygo1ax2e

// I try, fail:
const entries = await pb.collection('entries').getFullList({
  filter: `'features="${filters}"'`,
  expand: 'features'
});

```

Pocketbase returns this:

```
  url: 'https://pb.my-project.com/api/collections/entries/records?page=1&perPage=500&skipTotal=1&filter=%27features%3D%221pnq0aoygo1ax2e%22%27&expand=features',
  status: 400,
  response: {
    code: 400,
    message: 'Something went wrong while processing your request.',
    data: {}
  },
  isAbort: false,
  originalError: {
    url: 'https://pb.my-project.com/api/collections/entries/records?page=1&perPage=500&skipTotal=1&filter=%27features%3D%221pnq0aoygo1ax2e%22%27&expand=features',
    status: 400,
    data: {
      code: 400,
      message: 'Something went wrong while processing your request.',
      data: {}
    }
  }
}
```

I just don't see it. I am sure it's a very dumb mistake

**Comment by ganigeorgiev** [comment]
Remove the first wrapping quotes, aka. it must be:

```js
// or filter: pb.filter("features={:id}", { id: filters })
filter: `features="${filters}"`
```

_edit: I forgot to mention that you can get an empty list response if your client request doesn't satisfy the List API rule of the collection (see the notes in [API rules](https://pocketbase.io/docs/api-rules-and-filters/#api-rules))._

**Reply by realJogicodes** [reply]
Thank you. Now I get an empty list returned, the issue with the syntax was addressed. Such a dumb mistake, and I thought it would be. 

Now, regarding the API stuff, the empty list suggest to me that there is something else amiss as you mentioned. I had a look at the API rules document as well as at my own settings and I think it is correctly configured. I have view and list open to anyone for both parent collection and child relation so that should work, no? 

Curious

**Reply by ganigeorgiev** [reply]
If you are receiving an empty list then either your request doesn't satisfy the List API rule of your collection or your provided filter doesn't match any record.

In both cases there is nothing much I can help further with.

If you think that this is a PocketBase issue, please export your collections configuration as json (Admin UI > Settings > Export collections) and provide details how you are initializing the SDK and how your are sending the request.

**Reply by ganigeorgiev** [reply]
Side-note: if `features` is a multiple `relation` field then comparing it to a single id string will not work because it is represented as serialized json array. For this case you can use dot-nation to target explicitly the `id` field of the `relation` records, eg.:

```js
filter: `features.id ?= "YOUR_REL_ID"`
```

**Reply by realJogicodes** [reply]
This could be it. For now I solved it by removing the filter and then doing an extra operation on the returned data. The Ids are coming back and I can just compare. Not ideal but works for now.

 Will try your new approach tomorrow. How would the syntax work if I wanted to filter two or three instead of just one?

**Reply by ganigeorgiev** [reply]
_This is getting really confusing and I'm not sure I can help you further. If you have other questions please open a new Q&A discussion with clear description on your collections structure and what you are trying to do to avoid the tedious back-and-forth (see [XY problem](https://xyproblem.info/))._

> How would the syntax work if I wanted to filter two or three instead of just one?

You can combine them with `||` (OR) condition, or progmattically this could like:

```js
var searchIds = ["id1", "id2", "id3"]

const entries = await pb.collection('entries').getFullList({
  filter: searchIds.map((id) => pb.filter("features.id ?= {:id}", { id })).join("||")
  expand: 'features'
});
```

**Reply by realJogicodes** [reply]
Sure, I won't ask further, taking the hint. 
Thanks for the syntax. If I run into more troubles, I will ask fresh
